### PR TITLE
overlord/ifacestate: use remapper when checking if system snap is installed

### PR DESCRIPTION
**Pull request by stolowski**
Rename ensureSystemSnapIsPresent to checkSystemSnapIsPresent and use remapper to find system snap name and make sure it's installed, instead of just `CoreInfo()`.

### CODE REVIEW

**Comment by mvo5** [comment]
Do we need a test for this?

File: overlord/ifacestate/helpers.go [hunk 1]
```diff
@@ -777,10 +777,12 @@ func connectDisconnectAffectedSnaps(t *state.Task) ([]string, error) {
 	return []string{plugRef.Snap, slotRef.Snap}, nil
 }
 
-func ensureSystemSnapIsPresent(st *state.State) error {
+func checkSystemSnapIsPresent(st *state.State) error {
+	// "system" gets remapped to either snapd or a core snap.
+	systemSnap := mapper.RemapSnapFromRequest("system")
```
Comment (pedronis) [inline overlord/ifacestate/helpers.go]: wondering if it's worth introducing mapper.SystemSnapName() to mean this?

Comment (stolowski) [reply overlord/ifacestate/helpers.go]: Will do in a followup.

Comment (stolowski) [reply overlord/ifacestate/helpers.go]: Added with #6146

**Pull request merged**